presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -1054,9 +1056,11 @@ public synchronized void grantTablePrivileges(String databaseName, String tableN
     public synchronized void revokeTablePrivileges(String databaseName, String tableName, String tableOwner, HivePrincipal grantee, Set<HivePrivilegeInfo> privileges)
     {
         Set<HivePrivilegeInfo> currentPrivileges = listTablePrivileges(databaseName, tableName, tableOwner, Optional.of(grantee));
-        currentPrivileges.removeAll(privileges);
+        Set<HivePrivilegeInfo> privilegesToRemove = privileges.stream()
+                .map(p -> new HivePrivilegeInfo(p.getHivePrivilege(), p.isGrantOption(), p.getGrantor(), grantee))
+                .collect(toUnmodifiableSet());
 
-        setTablePrivileges(grantee, databaseName, tableName, currentPrivileges);
+        setTablePrivileges(grantee, databaseName, tableName, Sets.difference(currentPrivileges, privilegesToRemove));
```

Comment:
Please fix file based metastore in separate commit

Reply:
will split into 3 commits a moment before merge, doing this on every fix is clumsy

Reply:
> will split into 3 commits a moment before merge

A person who merge and so accepts the code has to see the code in a form it is going to be merged.

Anyway, I think it could stay as it is now. Single commit is not perfect here, but it is good enough.

Reply:
done